src/OrchardCore/Orchard.Localization.Core/StringLocalizer.cs

```diff
@@ -74,25 +78,45 @@ public IEnumerable<LocalizedString> GetAllStrings(bool includeParentCultures)
 
         public IStringLocalizer WithCulture(CultureInfo culture)
         {
-            return new PoStringLocalizer(culture, Context, _localizationManager);
+            return new StringLocalizer(culture, Context, _localizationManager, _logger);
+        }
+
+        private string GetTranslation(string[] pluralForms, int? count)
+        {
+            var pluralForm = count.HasValue ? _dictionary.PluralRule(count.Value) : 0;
+            if (pluralForm >= pluralForms.Length)
+            {
+                throw new PluralFormNotFoundException($"Plural form '{pluralForm}' doesn't exist in values provided by the IStringLocalizer.Plural method. Provided values: {pluralForms}");
+            }
+
+            return pluralForms[pluralForm];
         }
 
         private string GetTranslation(string name, string context, int? count)
         {
             var key = CultureDictionaryRecord.GetKey(name, context);
-            var translation = _dictionary[key, count];
-
-            if (translation == null && _parentCultureDictionary != null)
+            try
             {
-                translation = _parentCultureDictionary[key, count]; // fallback to the parent culture
-            }
+                var translation = _dictionary[key, count];
+
+                if (translation == null && _parentCultureDictionary != null)
+                {
+                    translation = _parentCultureDictionary[key, count]; // fallback to the parent culture
+                }
 
-            if (translation == null && context != null)
+                if (translation == null && context != null)
+                {
+                    translation = GetTranslation(name, null, count); // fallback to the translation without context
+                }
+
+                return translation;
+            }
+            catch (PluralFormNotFoundException ex)
```

Comment:
I see, so this exception is not exposed to the end user. In this case don't remove it as I suggested.